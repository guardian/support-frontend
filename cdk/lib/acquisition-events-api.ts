import { DomainName, HttpApi } from "@aws-cdk/aws-apigatewayv2-alpha";
import { GuAlarm } from "@guardian/cdk/lib/constructs/cloudwatch";
import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
import { GuStack } from "@guardian/cdk/lib/constructs/core";
import type { App } from "aws-cdk-lib";
import { Duration } from "aws-cdk-lib";
import { CfnIntegration, CfnRoute } from "aws-cdk-lib/aws-apigatewayv2";
import { Certificate } from "aws-cdk-lib/aws-certificatemanager";
import { ComparisonOperator, Metric } from "aws-cdk-lib/aws-cloudwatch";
import {
  Effect,
  PolicyStatement,
  Role,
  ServicePrincipal,
} from "aws-cdk-lib/aws-iam";
import { CfnRecordSet } from "aws-cdk-lib/aws-route53";

export interface AcquisitionEventsApiProps extends GuStackProps {
  stack: string;
  stage: string;
  certificateId: string;
  domainName: string;
  hostedZoneName: string;
  hostedZoneId: string;
}

export class AcquisitionEventsApi extends GuStack {
  constructor(scope: App, id: string, props: AcquisitionEventsApiProps) {
    super(scope, id, props);

    const app = "acquisition-events-api";

    // ---- Alarms ---- //
    const alarmName = (shortDescription: string) =>
      `ACQUISITION-EVENTS-API-CDK- ${this.stage} ${shortDescription}`;

    const alarmDescription = (description: string) =>
      `Impact - ${description}. Follow the process in https://docs.google.com/document/d/1_3El3cly9d7u_jPgTcRjLxmdG2e919zCLvmcFCLOYAk/edit`;

    new GuAlarm(this, "ApiGateway4XXAlarmCDK", {
      app,
      alarmName: alarmName("API gateway 4XX response"),
      alarmDescription: alarmDescription(
        "Acquisition Events API received an invalid request"
      ),
      evaluationPeriods: 1,
      threshold: 1,
      snsTopicName: 'alarms-handler-topic-PROD',
      actionsEnabled: this.stage === "PROD",
      comparisonOperator: ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
      metric: new Metric({
        metricName: "4XXError",
        namespace: "AWS/ApiGateway",
        statistic: "Sum",
        period: Duration.seconds(300),
        dimensionsMap: {
          ApiName: `${app}-${this.stage}`,
        },
      }),
    });

    // ---- DNS ---- //
    const certificateArn = `arn:aws:acm:eu-west-1:${this.account}:certificate/${props.certificateId}`;

    const domainName = new DomainName(this, "DomainName", {
      domainName: props.domainName,
      certificate: Certificate.fromCertificateArn(this, "cert", certificateArn),
    });

    new CfnRecordSet(this, "DNSRecord", {
      name: props.domainName,
      type: "CNAME",
      hostedZoneId: props.hostedZoneId,
      ttl: "120",
      resourceRecords: [domainName.regionalDomainName],
    });

    // ---- Eventbridge stuff ---- //
    // There's no Eventbridge integration available as CDK L2 yet, so we have to use L1 and create Role, Integration and Route
    // https://docs.aws.amazon.com/cdk/api/v2/docs/aws-apigatewayv2-alpha-readme.html#defining-http-apis

    const eventBusArn = `arn:aws:events:eu-west-1:865473395570:event-bus/acquisitions-bus-${props.stage}`;

    // Api
    const httpApi = new HttpApi(
      this,
      `acquisitions-eventbridge-api-${props.stage}`,
      {
        defaultDomainMapping: {
          domainName: domainName,
        },
      }
    );

    // Role
    const apiRole = new Role(this, "EventBridgeIntegrationRole", {
      assumedBy: new ServicePrincipal("apigateway.amazonaws.com"),
    });

    apiRole.addToPolicy(
      new PolicyStatement({
        effect: Effect.ALLOW,
        resources: [eventBusArn],
        actions: ["events:PutEvents"],
      })
    );

    // Integration
    const eventBridgeIntegration = new CfnIntegration(
      this,
      "EventBridgeIntegration",
      {
        apiId: httpApi.httpApiId,
        integrationType: "AWS_PROXY",
        integrationSubtype: "EventBridge-PutEvents",
        credentialsArn: apiRole.roleArn,
        requestParameters: {
          Source: "ApiGateway",
          DetailType: "AcquisitionsEvent",
          Detail: "$request.body",
          EventBusName: eventBusArn,
        },
        payloadFormatVersion: "1.0",
        timeoutInMillis: 10000,
      }
    );

    // Routes
    new CfnRoute(this, "AcquisitionRoute", {
      apiId: httpApi.httpApiId,
      routeKey: "POST /acquisition",
      target: `integrations/${eventBridgeIntegration.ref}`,
    });

    new CfnRoute(this, "FallbackRoute", {
      apiId: httpApi.httpApiId,
      routeKey: "POST /",
      target: `integrations/${eventBridgeIntegration.ref}`,
    });
  }
}
