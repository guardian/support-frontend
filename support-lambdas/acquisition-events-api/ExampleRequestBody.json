{
  "eventTimeStamp": "2018-12-13T14:15:04.165Z",
  "product": "RECURRING_CONTRIBUTION",
  "amount": 1,
  "country": "GB",
  "currency": "BHD",
  "componentId": "MY_COMPONENT_ID",
  "componentType": null,
  "campaignCode": "MY_CAMPAIGN_CODE",
  "source": null,
  "referrerUrl": "referrer-url",
  "abTests": [],
  "paymentFrequency": "MONTHLY",
  "paymentProvider": "STRIPE",
  "printOptions": null,
  "browserId": null,
  "identityId": null,
  "pageViewId": null,
  "referrerPageViewId": null,
  "labels": [],
  "promoCode": null,
  "reusedExistingPaymentMethod": false,
  "readerType": "Direct",
  "acquisitionType": "Purchase",
  "zuoraSubscriptionNumber": null,
  "contributionId": null,
  "paymentId": null,
  "queryParameters": [
    {
      "name": "name1",
      "value": "value1"
    },
    {
      "name": "name2",
      "value": "value2"
    }
  ],
  "platform": null,
  "postalCode": null,
  "state": null,
  "email": null
}
