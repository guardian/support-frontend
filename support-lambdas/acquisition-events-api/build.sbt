import LibraryVersions.circeVersion
import LibraryVersions._

name := "acquisition-events-api"
description := "A lambda for acquisitions events api"

libraryDependencies ++= Seq(
  "io.circe" %% "circe-core" % circeVersion,
  "io.circe" %% "circe-generic" % circeVersion,
)

assemblyJarName := s"${name.value}.jar"
riffRaffPackageType := assembly.value
riffRaffUploadArtifactBucket := Option("riffraff-artifact")
riffRaffUploadManifestBucket := Option("riffraff-builds")
riffRaffManifestProjectName := s"support:lambdas:${name.value}"
riffRaffArtifactResources += (file(
  "cdk/cdk.out/Acquisition-Events-API-PROD.template.json",
), "cfn/Acquisition-Events-API-PROD.template.json")
riffRaffArtifactResources += (file(
  "cdk/cdk.out/Acquisition-Events-API-CODE.template.json",
), "cfn/Acquisition-Events-API-CODE.template.json")
