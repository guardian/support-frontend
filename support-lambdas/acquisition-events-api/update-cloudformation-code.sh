#!/bin/sh

cd ../..

pnpm --filter cdk synth # If this fails check you are using the correct node version

# Exit if any of these commands fail, print commands to console
set -ex

aws cloudformation update-stack \
  --capabilities '["CAPABILITY_AUTO_EXPAND", "CAPABILITY_NAMED_IAM", "CAPABILITY_IAM"]'  \
  --stack-name support-CODE-acquisition-events-api \
  --template-body file://cdk.out/Acquisition-Events-Api-CODE.template.json \
  --profile membership

cd -
echo -e "\nStack update has been started, check progress in the AWS console.";
echo -e "https://eu-west-1.console.aws.amazon.com/cloudformation/home";
