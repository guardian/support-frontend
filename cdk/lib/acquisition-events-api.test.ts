import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { AcquisitionEventsApi } from "./acquisition-events-api";

describe("The Acquisition Events API stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const codeStack = new AcquisitionEventsApi(
      app,
      "Acquisition-Events-API-CODE",
      {
        stack: "support",
        stage: "CODE",
        certificateId: "b384a6a0-2f54-4874-b99b-96eeff96c009",
        domainName: "acquisition-events-code.support.guardianapis.com",
        hostedZoneName: "support.guardianapis.com.",
        hostedZoneId: "Z3KO35ELNWZMSX",
      }
    );
    const prodStack = new AcquisitionEventsApi(
      app,
      "Acquisition-Events-API-PROD",
      {
        stack: "support",
        stage: "PROD",
        certificateId: "b384a6a0-2f54-4874-b99b-96eeff96c009",
        domainName: "acquisition-events.support.guardianapis.com",
        hostedZoneName: "support.guardianapis.com.",
        hostedZoneId: "Z3KO35ELNWZMSX",
      }
    );

    expect(Template.fromStack(codeStack).toJSON()).toMatchSnapshot();
    expect(Template.fromStack(prodStack).toJSON()).toMatchSnapshot();
  });
});
