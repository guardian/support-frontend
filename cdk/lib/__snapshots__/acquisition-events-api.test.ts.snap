// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Acquisition Events API stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuAlarm",
    ],
    "gu:cdk:version": "TEST",
  },
  "Resources": {
    "AcquisitionRoute": {
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiCODE8060A47C",
        },
        "RouteKey": "POST /acquisition",
        "Target": {
          "Fn::Join": [
            "",
            [
              "integrations/",
              {
                "Ref": "EventBridgeIntegration",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "ApiGateway4XXAlarmCDKC83EACCA": {
      "Properties": {
        "ActionsEnabled": false,
        "AlarmActions": [
          {
            "Fn::Join": [
              "",
              [
                "arn:aws:sns:",
                {
                  "Ref": "AWS::Region",
                },
                ":",
                {
                  "Ref": "AWS::AccountId",
                },
                ":alarms-handler-topic-PROD",
              ],
            ],
          },
        ],
        "AlarmDescription": "Impact - Acquisition Events API received an invalid request. Follow the process in https://docs.google.com/document/d/1_3El3cly9d7u_jPgTcRjLxmdG2e919zCLvmcFCLOYAk/edit",
        "AlarmName": "ACQUISITION-EVENTS-API-CDK- CODE API gateway 4XX response",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": [
          {
            "Name": "ApiName",
            "Value": "acquisition-events-api-CODE",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "4XXError",
        "Namespace": "AWS/ApiGateway",
        "Period": 300,
        "Statistic": "Sum",
        "Threshold": 1,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "DNSRecord": {
      "Properties": {
        "HostedZoneId": "Z3KO35ELNWZMSX",
        "Name": "acquisition-events-code.support.guardianapis.com",
        "ResourceRecords": [
          {
            "Fn::GetAtt": [
              "DomainNameEC95A6E9",
              "RegionalDomainName",
            ],
          },
        ],
        "TTL": "120",
        "Type": "CNAME",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "DomainNameEC95A6E9": {
      "Properties": {
        "DomainName": "acquisition-events-code.support.guardianapis.com",
        "DomainNameConfigurations": [
          {
            "CertificateArn": {
              "Fn::Join": [
                "",
                [
                  "arn:aws:acm:eu-west-1:",
                  {
                    "Ref": "AWS::AccountId",
                  },
                  ":certificate/b384a6a0-2f54-4874-b99b-96eeff96c009",
                ],
              ],
            },
            "EndpointType": "REGIONAL",
          },
        ],
        "Tags": {
          "Stack": "support",
          "Stage": "CODE",
          "gu:cdk:version": "TEST",
          "gu:repo": "guardian/support-frontend",
        },
      },
      "Type": "AWS::ApiGatewayV2::DomainName",
    },
    "EventBridgeIntegration": {
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiCODE8060A47C",
        },
        "CredentialsArn": {
          "Fn::GetAtt": [
            "EventBridgeIntegrationRoleB322760B",
            "Arn",
          ],
        },
        "IntegrationSubtype": "EventBridge-PutEvents",
        "IntegrationType": "AWS_PROXY",
        "PayloadFormatVersion": "1.0",
        "RequestParameters": {
          "Detail": "$request.body",
          "DetailType": "AcquisitionsEvent",
          "EventBusName": "arn:aws:events:eu-west-1:865473395570:event-bus/acquisitions-bus-CODE",
          "Source": "ApiGateway",
        },
        "TimeoutInMillis": 10000,
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "EventBridgeIntegrationRoleB322760B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-frontend",
          },
          {
            "Key": "Stack",
            "Value": "support",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "EventBridgeIntegrationRoleDefaultPolicy16371A00": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "events:PutEvents",
              "Effect": "Allow",
              "Resource": "arn:aws:events:eu-west-1:865473395570:event-bus/acquisitions-bus-CODE",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventBridgeIntegrationRoleDefaultPolicy16371A00",
        "Roles": [
          {
            "Ref": "EventBridgeIntegrationRoleB322760B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "FallbackRoute": {
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiCODE8060A47C",
        },
        "RouteKey": "POST /",
        "Target": {
          "Fn::Join": [
            "",
            [
              "integrations/",
              {
                "Ref": "EventBridgeIntegration",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "acquisitionseventbridgeapiCODE8060A47C": {
      "Properties": {
        "Name": "acquisitions-eventbridge-api-CODE",
        "ProtocolType": "HTTP",
        "Tags": {
          "Stack": "support",
          "Stage": "CODE",
          "gu:cdk:version": "TEST",
          "gu:repo": "guardian/support-frontend",
        },
      },
      "Type": "AWS::ApiGatewayV2::Api",
    },
    "acquisitionseventbridgeapiCODEDefaultStage8CFF60B0": {
      "DependsOn": [
        "DomainNameEC95A6E9",
      ],
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiCODE8060A47C",
        },
        "AutoDeploy": true,
        "StageName": "$default",
        "Tags": {
          "Stack": "support",
          "Stage": "CODE",
          "gu:cdk:version": "TEST",
          "gu:repo": "guardian/support-frontend",
        },
      },
      "Type": "AWS::ApiGatewayV2::Stage",
    },
    "acquisitionseventbridgeapiCODEDefaultStageAcquisitionEventsAPICODEDomainNameundefined4F0D5A11": {
      "DependsOn": [
        "acquisitionseventbridgeapiCODEDefaultStage8CFF60B0",
        "DomainNameEC95A6E9",
      ],
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiCODE8060A47C",
        },
        "DomainName": {
          "Ref": "DomainNameEC95A6E9",
        },
        "Stage": "$default",
      },
      "Type": "AWS::ApiGatewayV2::ApiMapping",
    },
  },
}
`;

exports[`The Acquisition Events API stack matches the snapshot 2`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuAlarm",
    ],
    "gu:cdk:version": "TEST",
  },
  "Resources": {
    "AcquisitionRoute": {
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiPROD575D58B4",
        },
        "RouteKey": "POST /acquisition",
        "Target": {
          "Fn::Join": [
            "",
            [
              "integrations/",
              {
                "Ref": "EventBridgeIntegration",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "ApiGateway4XXAlarmCDKC83EACCA": {
      "Properties": {
        "ActionsEnabled": true,
        "AlarmActions": [
          {
            "Fn::Join": [
              "",
              [
                "arn:aws:sns:",
                {
                  "Ref": "AWS::Region",
                },
                ":",
                {
                  "Ref": "AWS::AccountId",
                },
                ":alarms-handler-topic-PROD",
              ],
            ],
          },
        ],
        "AlarmDescription": "Impact - Acquisition Events API received an invalid request. Follow the process in https://docs.google.com/document/d/1_3El3cly9d7u_jPgTcRjLxmdG2e919zCLvmcFCLOYAk/edit",
        "AlarmName": "ACQUISITION-EVENTS-API-CDK- PROD API gateway 4XX response",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": [
          {
            "Name": "ApiName",
            "Value": "acquisition-events-api-PROD",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "4XXError",
        "Namespace": "AWS/ApiGateway",
        "Period": 300,
        "Statistic": "Sum",
        "Threshold": 1,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "DNSRecord": {
      "Properties": {
        "HostedZoneId": "Z3KO35ELNWZMSX",
        "Name": "acquisition-events.support.guardianapis.com",
        "ResourceRecords": [
          {
            "Fn::GetAtt": [
              "DomainNameEC95A6E9",
              "RegionalDomainName",
            ],
          },
        ],
        "TTL": "120",
        "Type": "CNAME",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "DomainNameEC95A6E9": {
      "Properties": {
        "DomainName": "acquisition-events.support.guardianapis.com",
        "DomainNameConfigurations": [
          {
            "CertificateArn": {
              "Fn::Join": [
                "",
                [
                  "arn:aws:acm:eu-west-1:",
                  {
                    "Ref": "AWS::AccountId",
                  },
                  ":certificate/b384a6a0-2f54-4874-b99b-96eeff96c009",
                ],
              ],
            },
            "EndpointType": "REGIONAL",
          },
        ],
        "Tags": {
          "Stack": "support",
          "Stage": "PROD",
          "gu:cdk:version": "TEST",
          "gu:repo": "guardian/support-frontend",
        },
      },
      "Type": "AWS::ApiGatewayV2::DomainName",
    },
    "EventBridgeIntegration": {
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiPROD575D58B4",
        },
        "CredentialsArn": {
          "Fn::GetAtt": [
            "EventBridgeIntegrationRoleB322760B",
            "Arn",
          ],
        },
        "IntegrationSubtype": "EventBridge-PutEvents",
        "IntegrationType": "AWS_PROXY",
        "PayloadFormatVersion": "1.0",
        "RequestParameters": {
          "Detail": "$request.body",
          "DetailType": "AcquisitionsEvent",
          "EventBusName": "arn:aws:events:eu-west-1:865473395570:event-bus/acquisitions-bus-PROD",
          "Source": "ApiGateway",
        },
        "TimeoutInMillis": 10000,
      },
      "Type": "AWS::ApiGatewayV2::Integration",
    },
    "EventBridgeIntegrationRoleB322760B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-frontend",
          },
          {
            "Key": "Stack",
            "Value": "support",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "EventBridgeIntegrationRoleDefaultPolicy16371A00": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "events:PutEvents",
              "Effect": "Allow",
              "Resource": "arn:aws:events:eu-west-1:865473395570:event-bus/acquisitions-bus-PROD",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventBridgeIntegrationRoleDefaultPolicy16371A00",
        "Roles": [
          {
            "Ref": "EventBridgeIntegrationRoleB322760B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "FallbackRoute": {
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiPROD575D58B4",
        },
        "RouteKey": "POST /",
        "Target": {
          "Fn::Join": [
            "",
            [
              "integrations/",
              {
                "Ref": "EventBridgeIntegration",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ApiGatewayV2::Route",
    },
    "acquisitionseventbridgeapiPROD575D58B4": {
      "Properties": {
        "Name": "acquisitions-eventbridge-api-PROD",
        "ProtocolType": "HTTP",
        "Tags": {
          "Stack": "support",
          "Stage": "PROD",
          "gu:cdk:version": "TEST",
          "gu:repo": "guardian/support-frontend",
        },
      },
      "Type": "AWS::ApiGatewayV2::Api",
    },
    "acquisitionseventbridgeapiPRODDefaultStageAcquisitionEventsAPIPRODDomainNameundefined36D98266": {
      "DependsOn": [
        "acquisitionseventbridgeapiPRODDefaultStageEB205C3C",
        "DomainNameEC95A6E9",
      ],
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiPROD575D58B4",
        },
        "DomainName": {
          "Ref": "DomainNameEC95A6E9",
        },
        "Stage": "$default",
      },
      "Type": "AWS::ApiGatewayV2::ApiMapping",
    },
    "acquisitionseventbridgeapiPRODDefaultStageEB205C3C": {
      "DependsOn": [
        "DomainNameEC95A6E9",
      ],
      "Properties": {
        "ApiId": {
          "Ref": "acquisitionseventbridgeapiPROD575D58B4",
        },
        "AutoDeploy": true,
        "StageName": "$default",
        "Tags": {
          "Stack": "support",
          "Stage": "PROD",
          "gu:cdk:version": "TEST",
          "gu:repo": "guardian/support-frontend",
        },
      },
      "Type": "AWS::ApiGatewayV2::Stage",
    },
  },
}
`;
